streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -718,7 +718,7 @@ public class StreamsConfig extends AbstractConfig {
     /** {@code state.dir} */
     @SuppressWarnings("WeakerAccess")
     public static final String STATE_DIR_CONFIG = "state.dir";
-    private static final String STATE_DIR_DOC = "Directory location for state store. This path must be unique for each streams instance sharing the same underlying filesystem.";
+    private static final String STATE_DIR_DOC = "Directory location for state store. This path must be unique for each streams instance sharing the same underlying filesystem. Note that the location will be different in each environment as it is computed using System.getProperty(\"java.io.tmpdir\")";
```

Comment:
```suggestion
    private static final String STATE_DIR_DOC = "Directory location for state store. This path must be unique for each streams instance sharing the same underlying filesystem. Note that if not configured, then the default location will be different in each environment as it is computed using System.getProperty(\"java.io.tmpdir\")";
```
I think we should mention that only the default value depends on the output of `System.getProperty("java.io.tmpdir")`

Reply:
@machi1990 Thanks for pointing out! Yes you are right, and I've modified as suggested.